.github/workflows/update-e2e-template.yml

```diff
@@ -0,0 +1,103 @@
+name: Auto-update E2E template
+
+on:
+  workflow_call:
+    inputs:
+      os:
+        description: 'OS'
+        type: string
+        required: true
+      python-version:
+        description: 'Python version'
+        type: string
+        required: true
+
+  workflow_dispatch:
+    inputs:
+      os:
+        description: 'OS'
+        type: choice
+        options: 
+          - ubuntu-latest
+          - macos-latest
+          - windows-latest
+        required: false
+        default: 'ubuntu-latest'
+      python-version:
+        description: 'Python version'
+        type: choice
+        options:
+          - '3.8'
+          - '3.9'
+          - '3.10'
+          - '3.11'
+        required: false
+        default: '3.8'
+
+jobs:
+  update-e2e-template:
+    name: update-e2e-template
+    runs-on: ${{ inputs.os }}
```

Comment:
Will this run on multiple runners? If we're just updating + making a pull request, could it just run on a single runner?

Reply:
Pointed to code below